src/main/resources/db/changesets/20200625_initial_schema.yaml

```diff
@@ -0,0 +1,143 @@
+                nullable: true
+                unique: false
+    - createTable:
+        tableName: tracked_resource
+        columns:
+          - column:
+              name: tracked_id
+              type: uuid
+              constraints:
+                primaryKey: true
+                nullable: false
+          - column:
+              name: resource_type_id
+              type: uuid
+              constraints:
+                references: resource_type(type_id)
+                foreignKeyName: fk_type_id
+                nullable: true
+          - column:
+              name: resource_uid
+              type: json
+              constraints:
+                nullable: true
+                unique: false
+          - column:
+              name: creation
+              type: timestamp
+              constraints:
+                nullable: true
+                unique: false
+          - column:
+              name: expireation
+              type: timestamp
+              constraints:
+                nullable: true
+                unique: false
+          - column:
+              name: state
+              type: text
+              constraints:
+                nullable: true
+                unique: false
+          - column:
+              name: wait_until
+              type: timestamp
+              constraints:
+                nullable: true
+                unique: false
+    - createTable:
+        tableName: label
+        columns:
+          - column:
+              name: tracked_id
+              type: uuid
+              constraints:
+                references: tracked_resource(tracked_id)
+                foreignKeyName: fk_tracked_id
+                primaryKey: true
+                nullable: false
+          - column:
+              name: key
+              type: text
+              constraints:
+                nullable: true
+                unique: false
+          - column:
+              name: value
+              type: text
+              constraints:
+                nullable: true
+                unique: false
+    - createTable:
+        tableName: cleanup_log
+        columns:
+          - column:
+              name: tracked_id
+              type: uuid
+              constraints:
+                references: tracked_resource(tracked_id)
+                foreignKeyName: fk_tracked_id
+                primaryKey: true
+                nullable: false
+          - column:
+              name: logtime
+              type: timestamp
+              constraints:
+                nullable: true
+                unique: false
+          - column:
+              name: status
+              type: text
+              constraints:
+                nullable: true
+                unique: false
+          - column:
+              name: description
+              type: text
+              constraints:
+                nullable: true
+                unique: false
+    - createTable:
+        tableName: gcp_bucket
```

Comment:
Slightly concerned that this might indicate a proliferation of tables for different resource types across different providers, which feels like we might be doing something wrong.  I would feel a little better if it were just a `buckets` table that could store information for all cloud providers in one table, but even that I'd like to avoid if we can.

Reply:
Right, the gcp_bucket does sound like a cloud resource type, but I feel the current schema looks like a gcp specific thing: (project_id + bucket_name)? I feel this table may need some future discussion. I will check with Will once he is back.

Is there any better table name which includes "gcp specific" + "bucket" but doesn't sound like a resource type?

Reply:
So the gcp_bucket table was meant as a dummy example of a database schema where we did do a proliferation of tables for different resource types. The other option proposed in the doc was the `resource_uid` json field. We should not do both. We did not discuss this in any review of the doc: https://docs.google.com/document/d/13mYVJML_fOLsX1gxQxRJgECUNT27dAMKzEJxUEvtQqM/edit#heading=h.vhh24t1ac0cs
So options:
1. json field to store arbitrary cloud resource uid on `tracked_resource` table. 
2. proliferation of tables, one per resource type with nasty joins across many tables.
Third option wasn't in the doc, but is being done here:
3. table with union of all fields from all resource types, as is being started here. Every field can be nullable, we keep extending this table whenever we need to add a resource type with a new column name.
I think we're better off discussing this offline then in the comments here.

Reply:
Option 2 sounds too many tables will be needed and I'd prefer option 1 looks good to me.  
Reason: seems json query vs creates/maintains 10+tables json query is simpler and easier. A nice json query template is easier to implement (https://www.postgresqltutorial.com/postgresql-json/) compare with a template for updating/inserting/query across tables based on resource type? 

this pr has 1 and 2 which is not correct. Do we want this initial schema just have the 1st option(or 2 or none) for now and change later once we decide. Or do we want to hold on this pr and wait until decision being made?

Reply:
I vote for option 1 and let this PR move ahead. I prefer 1 to 2 or 3. If we change our minds, at some point in development, we can change our database.